docs/content/en/docs/Design/management.md

```diff
@@ -0,0 +1,176 @@
+in the same level could be hidden.
+
+Note that, as multiplication has two operands, the actual circuit for the latter
+two policies is `mult(ms(ct0), ms(ct1))`, whereas in the first policy the
+circuit is `ms(mult(ct0, ct1))`.
+
+The third policy has one more modulus switching than others, so it will need one
+more modulus.
+
+There are also other insertion strategy like inserting it dynamically based on
+current noise (see HElib) or lazy rescaling. Those are not implemented.
+
+### BGV: Scale management
+
+For the original BGV scheme, it is required to have `qi = 1 (mod t)` where `t`
+is the plaintext modulus. However in practice such requirement will make the
+choice of `qi` much scarce. So in the GHS variant, this condition is removed,
+with the price of scale management needed.
+
+Modulus switching from level `i` to level `i-1` is essentially dividing (with
+rounding) the ciphertext by `qi`, hence dividing the noise and payload message
+inside by `qi`. The message `m` can often be written as `[m]_t` as the
+computation happens in `Zt`, then by dividing of `qi` the result message would
+be `[m / qi]_t`.
+
+Note that when `qi = 1 (mod t)`, the result message is the same as the original
+message. However, in GHS variant such condition does not always hold, so we
+would call the `[q^{-1}]_t` part as the _scale_ of the message and compiler
+needs to record and manage it during computation. And when decrypting such scale
+must be removed to obtain the original message.
+
+Note that, for messages `m0` and `m1` of different scale `a` and `b`, we could
+not add them directly because `[a * m0 + b * m1]_t` is often not the expected
+result for user. Instead we need to adjust the scale of one message to match
+another so `[b * m0 + b * m1]_t = [b * (m0 + m1)]_t`. Such adjustment could be
+done by multiplying `m0` with a constant `[b / a]_t`. This adjustment is not for
+free as it will also amplify the noise.
+
+As one may expect, different modulus switching insertion strategy would affect
+the scale on the message. For `m0` with scale `a` and `input1` with scale `b`,
+the result scale would be
+
+1. After multiplication: `[ab / qi]_t`.
+
+1. Before multiplication: `[a / qi * b / qi]_t` = `[ab / (qi^2)]_t`.
+
+This is messy enough. To ease the burden, we can impose additional requirement:
+Mandate a constant scale `Delta_i` for each level. This is called
+_level-specific scaling factor_. With this in mind, addition within one level
+can happen without caring about the scale, and the scale of the next level is
+also the same.
+
+1. After multiplication: `Delta_{i-1} = [Delta_i^2 / qi]_t`
+
+1. Before multiplication: `Delta_{i-1} = [Delta_i^2 / (qi^2)]_t`
+
+### BGV: Cross Level Operation
+
+With the level-specific scaling factor, one may wonder how could addition and
+multiplication of ciphertext of different levels happen as they have different
+scaling factor. This can be solved by adjusting the level and scale of the
+ciphertext of higher level.
```

Comment:
```suggestion
With the level-specific scaling factor, one may wonder how to perform addition and
multiplication of ciphertexts on different levels. This can be done by adjusting
the level and scale of the ciphertext at the higher level.
```